src/Api/GenericApi.php

```diff
@@ -4,18 +4,49 @@
 
 namespace Sylius\PayPalPlugin\Api;
 
-use GuzzleHttp\ClientInterface;
+use GuzzleHttp\ClientInterface as GuzzleClientInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
 
 final class GenericApi implements GenericApiInterface
 {
-    private ClientInterface $client;
+    public function __construct(
+        private readonly GuzzleClientInterface|ClientInterface $client,
+        private readonly ?RequestFactoryInterface $requestFactory = null,
+    ) {
+        if ($this->client instanceof GuzzleClientInterface) {
+            trigger_deprecation(
+                'sylius/paypal-plugin',
+                '1.6',
+                'Passing GuzzleHttp\ClientInterface as a first argument in the constructor is deprecated and will be removed. Use Psr\Http\Client\ClientInterface instead.',
+            );
+        }
 
-    public function __construct(ClientInterface $client)
-    {
-        $this->client = $client;
+        if (null === $this->requestFactory) {
+            trigger_deprecation(
+                'sylius/paypal-plugin',
+                '1.6',
+                'Not passing $requestFactory to %s constructor is deprecated and will be removed',
+                self::class,
+            );
+        }
     }
 
     public function get(string $token, string $url): array
+    {
+        if ($this->client instanceof GuzzleClientInterface && null === $this->requestFactory) {
```

Comment:
What if the `$this->requestFactory` will be null but `$this->client` will not be instance of `GuzzleClientInterface`? You should cover more cases checking the arguments

Reply:
I've changed AND to OR